source/common/upstream/load_balancer_impl.cc

```diff
@@ -828,8 +857,24 @@ HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector
   return candidate_host;
 }
 
+HostConstSharedPtr RandomLoadBalancer::peekAnotherHost(LoadBalancerContext* context) {
```

Comment:
This is the same code as below other than the random param, right? Can there be a common function shared between them?